lib/internal/streams/writable.js

```diff
@@ -698,11 +698,7 @@ function callFinal(stream, state) {
       if (typeof then === 'function') {
         then.call(
           result,
-          function() {
-            if (!called) {
-              process.nextTick(onFinish, null);
-            }
-          },
+          nop,
```

Comment:
Why?

Reply:
My idea was that in a non-async implementation of `_final`, the user code could hold on to the `onFinish` callback until it wanted to actually finish the stream. With an `async _final`, the user loses this control and `onFinished` is always called after the promise resolves. 

Granted, this is a problem only when async/await is mixed with callbacks, which can be avoided with a simple `await new Promise(...)`, but it might still be considered "unexpected" when compared to the sync version.

However, after #40773  this is indeed useless.